src/main/java/seedu/address/logic/commands/EditMeetingCommand.java

```diff
@@ -0,0 +1,257 @@
+ */
+public class EditMeetingCommand extends Command {
+
+    public static final String COMMAND_WORD = "editm";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Edits the details of the meeting identified "
+            + "by the index number used in the displayed meeting list. "
+            + "Existing values will be overwritten by the input values.\n"
+            + "Parameters: INDEX (must be a positive integer) "
+            + "[" + PREFIX_NAME + "NAME] "
+            + "[" + PREFIX_LOCATION + "LOCATION] "
+            + "[" + PREFIX_START_TIME + "START TIME] "
+            + "[" + PREFIX_END_TIME + "END TIME] "
+            + "Example: " + COMMAND_WORD + " 1 "
+            + PREFIX_NAME + "John Doe "
+            + PREFIX_LOCATION + "Discussion Room 3 "
+            + PREFIX_START_TIME + "09-10-2024 13:00 "
+            + PREFIX_END_TIME + "09-10-2024 14:00";
+
+    public static final String MESSAGE_EDIT_MEETING_SUCCESS = "Edited Meeting: %1$s";
+    public static final String MESSAGE_MEETING_NOT_EDITED = "At least one field to edit must be provided.";
+    public static final String MESSAGE_DUPLICATE_MEETING = "This meeting already exists.";
+
+    private final Index index;
+    private final EditMeetingDescriptor editMeetingDescriptor;
+
+    /**
+     * @param index of the meeting in the meeting list to edit
+     * @param editMeetingDescriptor details to edit the meeting with
+     */
+    public EditMeetingCommand(Index index, EditMeetingDescriptor editMeetingDescriptor) {
+        requireNonNull(index);
+        requireNonNull(editMeetingDescriptor);
+
+        this.index = index;
+        this.editMeetingDescriptor = new EditMeetingDescriptor(editMeetingDescriptor);
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+
+        if (index.getZeroBased() >= model.getMeetingSize()) {
+            System.out.println("INDEX VALUE: " + index.getZeroBased() + ", MEETING SIZE: " + model.getMeetingSize());
+            throw new CommandException(Messages.MESSAGE_INVALID_MEETING_INDEX);
+        }
+
+        Meeting meetingToEdit = model.getMeeting(index.getZeroBased());
+        Meeting editedMeeting = createEditedMeeting(meetingToEdit, editMeetingDescriptor);
+
+        // Check if the new name exists or not
+        Name newName = editedMeeting.getPersonToMeet();
+        Person personBeingEdited = null;
+        for (Person person : model.getFilteredPersonList()) {
+            if (person.getName().equals(meetingToEdit.getPersonToMeet())) {
+                personBeingEdited = person;
+            }
+        }
+
+        Person personToEditMeeting = null;
+        boolean personExists = false;
+        for (Person person : model.getFilteredPersonList()) {
+            if (person.getName().equals(newName)) {
+                personExists = true;
+                personToEditMeeting = person;
```

Comment:
It's great that you are considering the potential bugs of editing the names in the meetings!